src/protocollogin.cpp

```diff
@@ -183,6 +184,12 @@ void ProtocolLogin::onRecvFirstMessage(NetworkMessage& msg)
 		return;
 	}
 
+	if (g_config.getBoolean(ConfigManager::MAINTENANCE_MODE)) {
+		disconnectClient("The Server is in maintenance mode and only players with positions above tutors can login.",
+		                 version);
+		return;
+	}
```

Comment:
This will block anyone from logging in, regardless of account type or CanAlwaysLogin flag

Reply:
Also, if anything, this shouldn't be done in protocollogin at all, it should be in protocolgame